### DISCUSSION TITLE
NettyClientTransportConfig$Builder NoClassDefFoundError

### DISCUSSION THREAD

**Discussion by nicola-cecchi** [digitalpetri/modbus#76] [Q&A]
I need to read from a Modbus TCP device.
I created a sample class using the quick start example of modbus-tcp. The snippet:

    package com.nicola.nicmodbus;

    import com.digitalpetri.modbus.client.NettyTcpClientTransport;
    import com.digitalpetri.modbus.client.NettyClientTransportConfig;

    public class NicModbus {

        public static void main(String[] args) {
            var transport = NettyTcpClientTransport.create(cfg -> {
                  cfg.hostname = "172.17.0.2";
                  cfg.port = 502;
            });
         }
    }

trows the error:

Exception in thread "main" java.lang.NoClassDefFoundError: com/digitalpetri/modbus/client/NettyClientTransportConfig$Builder

when I run the java file. Buildind the project (Netbeans IDE 15) is ok.

Best regards, Thanks.

**Comment by kevinherron** [comment]
You must not have all of the JAR files on your class path you "run the java file", whatever you mean by that.

**Reply by nicola-cecchi** [reply]
I simply added a dependency to my project in order to include classes of com.digitalpetri.modbus.client. If I don't,  import com.digitalpetri.modbus.client.NettyTcpClientTransport is not allowed. JAR files are added  automatically to Dependencies node.

The dependency screenshot is attached.
"Run the java file" means execute the code of the snippet. If I comment the NettyTcpClientTransport.create line run is ok, so I think there is something wrong in the create instruction.

Thank you

![dependency](https://github.com/user-attachments/assets/3c701f73-24df-47b4-b1d7-efb751a3e054)
![Project](https://github.com/user-attachments/assets/e980d20b-3d25-4f9e-9abd-7483176b8c41)

**Reply by kevinherron** [reply]
Like I said previously, I can't help you with Netbeans issues. Never used it, don't know anything about it.

**Reply by nicola-cecchi** [reply]
Thank you anyway for your support.
Best regards
